File: curier-service/src/main/java/com/gokhana/courierservice/service/OrderEventService.java
package com.gokhana.courierservice.service;

import com.gokhana.courierservice.bus.subscriber.KafkaMessageConsumerService;
import com.gokhana.courierservice.model.event.EventType;
import com.gokhana.courierservice.model.event.OrderEvent;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Random;

@Service
public class OrderEventService {

    private final Logger LOGGER = LoggerFactory.getLogger(KafkaMessageConsumerService.class);

    List<String> couriers = List.of("Ahmet","Mehmet","Gökhan","John","Leonardo","Jane","Elon");
    Random rand = new Random();

    public void assignCourier(OrderEvent orderEvent){
        if(orderEvent.getEventType() == EventType.CREATED){
            LOGGER.info("Courier: {} is assigned to Requested Order: {}",couriers.get(new Random().nextInt(couriers.size())), orderEvent.getOrder());
        }
    }
}

File: email-service/src/main/kotlin/com/gokhana/emailservice/service/OrderEventService.kt
package com.gokhana.emailservice.service

import com.gokhana.emailservice.model.event.EventType
import com.gokhana.emailservice.model.event.OrderEvent
import org.slf4j.Logger
import org.slf4j.LoggerFactory
import org.springframework.stereotype.Service

@Service
class OrderEventService {
    private val LOGGER: Logger = LoggerFactory.getLogger(OrderEventService::class.java)

    fun sendEmail(orderEvent: OrderEvent) {
        if (orderEvent.eventType === EventType.CREATED) {
            LOGGER.info(
                "Email is sending for ${orderEvent.order.userName} to ${orderEvent.order.email}"
            )
        }
    }
}